File: 2020_QR/QuickReaction_Android-develop/app/src/main/java/com/example/QuickReactionMJ/network/ApplicationController.java
package com.example.QuickReactionMJ.network;


import android.app.Application;
import android.util.Log;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class ApplicationController extends Application {
    private String baseUrl;

    private static ApplicationController instance = new ApplicationController();
    public static ApplicationController getInstance() {return instance;}
    private NetworkService networkService;

    @Override
    public void onCreate(){
        super.onCreate();
        buildNetworkService();
        ApplicationController.instance = this;
    }

    public ApplicationController(){
        buildNetworkService();
        ApplicationController.instance = this;
    }


    public NetworkService getNetworkService() {return networkService;}

    public void buildNetworkService(){
        synchronized (ApplicationController.class){
            if(networkService == null){
                Log.i("App","전");
                //기본게이트웨이
                baseUrl = "http://192.168.0.32:8080/"; //원하는 URL 작성
                Gson gson = new GsonBuilder()
                        .setDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
                        .create();
                GsonConverterFactory factory = GsonConverterFactory.create(gson);

                Retrofit retrofit = new Retrofit.Builder()
                        .baseUrl(baseUrl)
                        .addConverterFactory(factory)
                        .build();
                networkService = retrofit.create(NetworkService.class);
                Log.i("App","후");
            }
        }
    }
}

File: 2020_QR/QuickReaction_Android-develop/app/src/main/java/com/example/QuickReactionMJ/network/ApplicationController.kt
/*package com.example.user.alcohol_measurement


import android.app.Application
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import javax.xml.datatype.DatatypeConstants.SECONDS
import javax.xml.datatype.DatatypeConstants.MINUTES
import okhttp3.OkHttpClient
import java.util.concurrent.TimeUnit


class  ApplicationController  :  Application()  {
    private  val  baseURL  =  "http://117.16.244.58:8080/"
    lateinit var networkService: NetworkService

    companion object {
        lateinit var instance: ApplicationController
    }

    override fun onCreate() {
        super.onCreate()
        instance = this
        buildNetWork()
    }

    //서버요청시 타임아웃 시간을 바꿀수있음(오래걸릴경우)
    fun buildNetWork() {
        val okHttpClient = OkHttpClient.Builder()
            .connectTimeout(10, TimeUnit.MINUTES)
            .readTimeout(30, TimeUnit.SECONDS)
            .writeTimeout(15, TimeUnit.SECONDS)
            .build()

        val retrofit: Retrofit = Retrofit.Builder()
            .baseUrl(baseURL)
            .client(okHttpClient)
            .addConverterFactory(GsonConverterFactory.create())
            .build()
        networkService = retrofit.create(NetworkService::class.java)
    }
}

*/